src/primitives/block.h

```diff
@@ -86,14 +86,14 @@ class CBlock : public CBlockHeader
     CBlock(const CBlockHeader &header)
     {
         SetNull();
-        *((CBlockHeader*)this) = header;
+        *(static_cast<CBlockHeader*>(this)) = header;
```

Comment:
@benma `this` isn't `const` in the constructor:

> The type of this in a member function of class X is X* (pointer to X). If the member function is cv-qualified, the type of this is cv X* (pointer to identically cv-qualified X). Since constructors and destructors cannot be cv-qualified, the type of this in them is always X*, even when constructing or destroying a const object. ([cppreference.com](http://en.cppreference.com/w/cpp/language/this))